src/User.php

```diff
@@ -733,7 +733,8 @@ public static function unsetUndisclosedFields(&$fields)
         $user = new self();
         $can_see_token = Session::getLoginUserID() === $fields['id']
             || (
-                $user->can($fields['id'], UPDATE)
+                !array_key_exists('id', $fields)
```

Comment:
The token should not be viewable if we cannot get the user ID (`$can_see_token` should be false if `!array_key_exists('id', $fields)`). Indeed, we would not be able to validate read access to it.